### PR TITLE
Update GitHub Actions workflows to use setup-go@v5 and golangci-lint-action@v6; refactor error formatting and import statements

### PR DESCRIPTION
**Pull request by devhindo**
fixes issues with existing code

edit: solves: #151

### CODE REVIEW

**Comment by devhindo** [comment]
all tests are passing now except for this issue! https://github.com/meshery/meshery/issues/13024 @leecalcote

**Comment by devhindo** [comment]
an opinion about the mentioned issue would be really helpful

File: main.go [hunk 1]
```diff
@@ -120,9 +120,10 @@ func main() {
 	}
 }
 
-func isDebug() bool {
-	return os.Getenv("DEBUG") == "true"
-}
+// unused function
+//func isDebug() bool {
+//	return os.Getenv("DEBUG") == "true"
+//}
```
Comment (theBeginner86) [inline main.go:127]: Instead of commenting, lets remove this function if not in use
```suggestion

```

**Review by theBeginner86 — APPROVED** [review]
Looks good, thanks!

**Comment by theBeginner86** [comment]
This PR fixes this issue: https://github.com/meshery/meshery-cilium/issues/151

**Comment by theBeginner86** [comment]
> an opinion about the mentioned issue would be really helpful

In my opinion, all E2E workflow are failing because the pattern file that is being used to deploy the mesh is of former OAM type and doesn't follows the new MeshModel schema due to which Meshery server is unable to deploy the sample design and hence the subsequent assertions that are made in workflow fails.

This error message confirms that pattern apply failed https://github.com/meshery/meshery-cilium/actions/runs/12528133889/job/34942427822?pr=173#step:17:14

Also, the status of all resources in K8s here https://github.com/meshery/meshery-cilium/actions/runs/12528133889/job/34942427822#step:22:1 confirms that the service mesh didn't come up.

If we can update these pattern files https://github.com/meshery/meshery-cilium/blob/master/.github/install/deploy.yaml based on the Meshery Design schema then we should be able to fix the issue.

File: main.go (hunk 1 above)
Comment (devhindo) [reply main.go:127]: done!

**Review by leecalcote — APPROVED** [review]
Good!

**Pull request merged**